File: crates/flowrt-codegen/tests/golden/feedback_v2_cpp/expected/app/stubs/cpp/controller.cpp
// FlowRT 管理参考模板（C++）。可删除重建；复制到用户 app/ 后再修改。

#include "flowrt_app/runtime_shell.hpp"

#include <memory>

namespace {

class Controller final : public flowrt_app::ControllerInterface {
public:
    flowrt::Status on_tick(
        const flowrt::Latest<flowrt_app::State>& state,
        flowrt::Output<flowrt_app::Cmd>& cmd) override {
        (void)state;
        cmd.write(flowrt_app::Cmd{});
        return flowrt::ok();
    }
};

}  // namespace

namespace flowrt_user {

flowrt_app::App build_app() {
    return flowrt_app::App(std::make_unique<Controller>());
}

}  // namespace flowrt_user

File: crates/flowrt-codegen/tests/golden/feedback_loop_rust/expected/app/stubs/rust/controller.rs
// FlowRT 管理参考模板（Rust）。可删除重建；复制到用户 app/ 后再修改。

#[derive(Default)]
pub struct Controller;

impl flowrt_app::components::Controller for Controller {
    fn on_tick(
        &mut self,
        state: flowrt::Latest<'_, flowrt_app::messages::State>,
        cmd: &mut flowrt::Output<flowrt_app::messages::Cmd>,
    ) -> flowrt::Status {
        let _ = state;
        cmd.write(flowrt_app::messages::Cmd::default());
        flowrt::Status::Ok
    }
}

pub fn build_app() -> flowrt_app::runtime_shell::App {
    flowrt_app::runtime_shell::App::new(Box::new(Controller::default()))
}
